### PR TITLE
New Edge sidebar PWA developer documentation

### PR DESCRIPTION
**Pull request by captainbrosset**
This short page describes how developers can make their PWAs ready for the Microsoft Edge sidebar.

Preview page for review: https://review.learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/sidebar?branch=pr-en-us-2458

AB#43480776

### CODE REVIEW

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:207]: Can we add a section to get feedback from developers on Sidebar features they'd like to add/see for PWAs? Not sure we have a GitHub for PWA specific developer feedback. Something that was discussed, but we'd like feedback on before reconsidering was the ability to specify a start_url in the edge_side_panel manifest section.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:146]: This could also be for sites.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:21]: .. may be promoted. I imagine that eventually if we succeed, we will need to balance promotion with other recommendation criteria, e.g. user engagement with sites, etc.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:154]: As Hoch mentioned, client hint headers will not be in 112, but planned to be upcoming soon (Target 113). We could say, coming soon/next?

**Review by ngaitonde — CHANGES_REQUESTED** [review]
Left a few comments. Thank you, this is looking good overall!

**Comment by slightlyoff** [comment]
Hey folks, thanks for putting this together.

I have a few concerns.

First, where is the Explainer for this design? We should never build features like this without showing our work and planning to take developer feedback. The usual way to do that is an Explainer and public GH design discussion, beginning in our own explainer repo, and eventually moving to WICG.

Next, is this opt-in via OT? Is that planned for the future?

Lastly, on the design, we can easily predict that some developers will want to treat the sidebar as an accessory or additive UI. The ways we can adapt to that are:

 - Developers can be fed information about the navigation occurring for a URL destined for that and serve up different content inside their Service Worker. This design doesn't provide that data on the `Request` or in the `clients` list.
 - We could directly offer a different starting point (e.g., a `start_url` within the `edge_side_panel` member).
 - We can provide that information at runtime using existing mechanisms for detecting display mode (less good) and potentially suffer a refresh.
 - We can provide this information through a new, less unified path (which this proposal provides)

I'm not sure how we're justifying going with the fourth option. I'd expect an Explainer to outline the logic here and weigh the tradeoffs.

Excited to see this moving forward, but to avoid future delays, we need to make sure the design is using the well-trod discussion and feedback mechanisms that will let us eventually take a version of it through the Blink Launch Process (including TAG review).

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:207]: https://github.com/MicrosoftDocs/edge-developer/issues could be a reasonable place to direct Issues. Thanks for the pointer Hoch!

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:174]:
```suggestion
**Note**: we strongly recommend using User Agent Client Hints instead of the User Agent string. The User Agent string is an outdated mechanism for doing browser detection, and has website compatibility issues.
```
gentler yet still direct wording

File: microsoft-edge/progressive-web-apps-chromium/demo-pwas.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/demo-pwas.md:34]:
```suggestion
| Sidebar | PWAmp can be pinned to the sidebar in Microsoft Edge. | [Build Microsoft Edge sidebar PWAs](how-to/sidebar.md). |
```
GLOBAL in this table column: remove period?  Checking rendered...

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (captainbrosset) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:207]: We have the MSEdgeExplainers repo that can be used for user feedback about new features. I'd prefer if we could create a new explainer there to engage with people and collect their feedback rather than use the edge-developer docs repo.

**Comment by kypflug** [comment]
@captainbrosset Thanks for the great start on the docs here. Let’s be sure to huddle on this internally before publishing any public docs. We need to ensure we are addressing Alex’s points above and have a planful approach to rolling out the API proposal (explainer, blog, etc.). I’ll set something up for us next week.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (ngaitonde) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:207]: Agreed. Adolf created this explainer now.: https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/SidePanel/explainer.md.

**Comment by captainbrosset** [comment]
Here is a link to the explainer: https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/SidePanel/explainer.md

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (captainbrosset) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:154]: Since we're waiting for feedback on the explainer anyway, maybe by the time we're ready to land the docs, the new CH headers will be available.
If they're not, I'll remove this part of the docs. It's better to document only what's available than to say some other pieces will come soon.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (captainbrosset) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:146]: Not sure what to do here. This documentation (and the entire parent area) is focused on PWA apps only. So rephrasing to "Detect when your app or site runs in the sidebar" feels a bit out of place here.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (adolfdaniel) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:176]:
```suggestion
If you can't use User Agent Client Hints in your app, look for `Edge Side Panel` in the User Agent string. Here are some examples of User Agent strings containing the `Edge Side Panel` value:
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (adolfdaniel) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:179]: same below

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (adolfdaniel) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:154]: This is being enabled through `msAddEdgeSidePaneBrandUA` feature flag.

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (sohchatt) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:146]: Let's use "Detect usage in the sidebar" then

Comment (sohchatt) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:146]: Detect usage in the sidebar

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (sohchatt) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:11]: Build PWAs for the sidebar in Microsoft Edge

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (sohchatt) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:29]: Experimental support for PWAs that can target the sidebar in Microsoft Edge starts with ...

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (sohchatt) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:53]: This is wrong. Should be edge_side_panel

Comment (sohchatt) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:53]: Here and everywhere else

**Comment by captainbrosset** [comment]
@sohchatt I believe I've addressed your comments. Please take a look when you can.

**Review by sohchatt — APPROVED** [review]

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:112]:
```suggestion
* Users can resize the sidebar to make it larger than your preferred width, or smaller, up to the 376 pixels minimum width.
```

Comment (mikehoffms) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:112]: this sentence could be smoother but i'll look at others before trying to draft alternatives

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:3]:
```suggestion
description: How to build your PWA for installation in the Microsoft Edge sidebar.
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:67]:
```suggestion
      "description": "A skinnable music player app to play your favorite mp3 files",
```
https://www.bing.com/search?pglt=129&q=skinable -- 309k hits if include skinnable, drops to 65k if strictly skinable

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:130]:
```suggestion
  "description": "A skinnable music player app to play your favorite mp3 files",
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:122]:
```suggestion
However, if you prefer to build an app that can only be pinned to the sidebar but cannot be not installed as a standalone app, omit the `display` member from your web app manifest, or set its value to `browser`:
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:148]:
```suggestion
Detecting when your app runs in the sidebar can be useful to provide your users with the best possible experience when your app runs in this specific browser UI.
```
does adding "surface" give a net gain, or introduces as much unclarity as clarification?

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:189]:
```suggestion
Suppose you create a desktop variant of your app and a mobile variant of your app.
```
"You might have created ...." is awkward and it sounds like speculating about user (avoid).

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:191]:
```suggestion
Suppose you create a desktop variant of your app and a mobile variant of your app. In this scenario, the desktop variant is used when your app is pinned to the sidebar. By default, apps in the sidebar that use the `edge_side_panel` manifest member receive the Microsoft Edge desktop User Agent Client Hint:
```
"You might have created ...." is awkward and it sounds like speculating about user (avoid).
Simplified wording to make the concept seem simple and familiar, not requiring more precision.
removed blank line

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:214]:
```suggestion
1. Open Microsoft Edge and make sure the sidebar is displayed. If the sidebar is not displayed, go to `edge://settings/sidebar` and then turn on the **Always show sidebar** toggle:
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/toggle
https://learn.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui - find 'toggle'

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:35]:
```suggestion
1. Start Microsoft Edge with sidebar support for PWAs, as follows:
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:41]:
```suggestion
        1. Find the path where Microsoft Edge Canary is installed on your device. For example: `C:\Users\your_name\AppData\Local\Microsoft\Edge SxS\Application\msedge.exe`.
        1. Enter **cmd** in the Windows search text box, and then select the **Command Prompt** app.
        1. Enter `"C:\Users\your_name\AppData\Local\Microsoft\Edge SxS\Application\msedge.exe" --enable-features=msWebAppManifestSidePanel,msAddEdgeSidePaneBrandUA`.
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:47]:
```suggestion
        1. Find the path where Microsoft Edge Canary is installed on your device. For example: `/Applications/Microsoft Edge Canary.app`.
        1. Open the **Terminal** app.
        1. Enter `/Applications/Microsoft Edge Canary.app --enable-features=msWebAppManifestSidePanel,msAddEdgeSidePaneBrandUA`.
```

Comment (mikehoffms) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:47]: result according to VS Code's preview with Learn pack:
image

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [reply microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:35]: reworked list section b/c "either" combined with unordered lists was confusing to me

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:83]:
```suggestion
The sidebar in Microsoft Edge has a default minimum width of 376 pixels and can be resized by users. Therefore, your app's layout should support the 376 pixels minimum width, and should be responsive.
```
separated the two thoughts

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:85]:
```suggestion
While building your app, you can test whether the app's layout is usable when narrow, and make sure the design is responsive at different widths, by using the **Device Emulation** tool in Microsoft Edge DevTools. To learn more, see [Verify that the webpage layout is usable when narrow](../../devtools-guide-chromium/accessibility/narrow.md).
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:120]:
```suggestion
The `edge_side_panel` manifest member makes your app installable as a standalone app and enables your app to be pinned to the sidebar in Microsoft Edge.
```

File: microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md
Comment (mikehoffms) [inline microsoft-edge/progressive-web-apps-chromium/how-to/sidebar.md:172]:
```suggestion
To detect whether your app is running in the sidebar, you can also use the User Agent string.
```

**Review by mikehoffms — CHANGES_REQUESTED** [review]
did an add'l Writer/Editor pass

**Review by Reezaali — APPROVED** [review]
Approving to allow the merge to go ahead when needed. There are a couple of comments to be resolved before merging to main please. Thanks!

**Comment by captainbrosset** [comment]
Addressed all review comments. Thanks all for the help here.

**Pull request merged**
